hw4/routes/index.js

```diff
@@ -0,0 +1,99 @@
+var mongoose = require('mongoose');
+var Ingredients = require('./../models/ingredientmodel');
+var routes = {};
+
+routes.home = function(req, res) {
+	res.render('home');
+};
+
+routes.edit = function(req, res) {
+	var updated = req.body;
+	Ingredients.update({'_id':updated.id}, updated, function(err, num, data) {
+		res.end(JSON.stringify(updated));
```

Comment:
`res.end()` will close the connection without sending any data back to the client. That'll make your jquery waiting for a response very sad. Use `res.send()` or `res.json()` instead to send data back.